docs/adr/0002-custom-dev-container-image.md

```diff
@@ -0,0 +1,34 @@
+# 2. Provide a custom dev container image
+
+Date: 2024-06-04
+
+## Status
+
+Accepted
+
+## Context
+
+We are using [dev containers](https://code.visualstudio.com/docs/devcontainers/create-dev-container) to provide a configuration for Codespaces that gives researchers a useful development environment to use while working on their OpenSAFELY projects. We want to improve that environment to make it more appealing and useful to researchers.
+
+## Decision
+
+We will have a single dev container environment with a custom Docker image, defined within the [research-template](https://github.com/opensafely/research-template/blob/main/.devcontainer/devcontainer.json) repo, available for researchers to use within Codespaces.
+
+We will use rocker/rstudio:4.0.5 as the base image, at least initially.
+
+We will copy relevant files into the Docker image, such as the Python and R packages from the Python and R action images.
+
+## Consequences
+
+We originally had a single dev container environment with a generic Docker image. However, we want to build on this to provide a more "habitable" environment for researchers in which to write code. This includes providing RStudio, as the majority of researchers using OpenSAFELY write R code. It also includes having the same Python and R packages available within the development environment, as they would find in the action images. This gives researchers syntax highlighting, help pages/documentation, and autocomplete. It also provides some ability to run code from within VS Code or R Studio, although we don't expect it will replace the use of `opensafely run` or `opensafely exec`.
+
+We believe having a single dev container environment better matches the researcher’s mental model of how the system works and is how they expect to work. It also provides a less steep learning curve. An alternative option was to have separate python and R dev container configurations and Docker images. However:
```

Comment:
I don't think a standard development environment matches researchers' mental models of how OpenSAFELY works. If it does, then they are going to struggle to use OpenSAFELY. I'd like to know more.

FWIW, I think a standard development environment helps researchers bridge the gap between how they worked in the past and how they will work in the present, with OpenSAFELY.

Reply:
Yes, I think does a better job of matching how researchers have previously worked and expect to work. I think rewording this section has bypassed/resolved the issue, but let me know if not.